### PR TITLE
Cleanup beforeSnapshot/ afterRestore

### PR DESCRIPTION
**Pull request by uweschaefer**

### CODE REVIEW

**Review by DanielGronau — APPROVED** [review]

**Comment by uweschaefer** [comment]
added some docs, @DanielGronau please take another look

**Comment by uweschaefer** [comment]
also renamed to **on**BeforeSnapshot() etc. to comply with naming patterns...

File: factcast-factus/src/main/java/org/factcast/factus/FactusImpl.java [hunk 1]
```diff
@@ -298,7 +301,14 @@ void createSnapshot(SnapshotProjection projection, UUID state) {
...
     Optional<Snapshot> latest = aggregateSnapshotRepository.findLatest(aggregateClass, aggregateId);
-    Optional<A> optionalA = latest.map(as -> ser.deserialize(aggregateClass, as.bytes()));
+    Optional<A> optionalA =
+        latest
+            .map(as -> ser.deserialize(aggregateClass, as.bytes()))
+            .map(
+                s -> {
+                  s.onAfterRestore();
+                  return s;
+                });
```
Comment (DanielGronau) [inline factcast-factus/src/main/java/org/factcast/factus/FactusImpl.java:311]: This is annoying, I saw the suggestion to have a utility method to simulate the behavior of `Stream.peek()`:
```
<T> UnaryOperator<T> peek(Consumer<T> c) {
    return x -> {
        c.accept(x);
        return x;
    };
}
```
Then we can write in such cases `...map(peek(s -> s.onAfterRestore()))....`

**Review by DanielGronau — APPROVED** [review]

File: factcast-factus/src/main/java/org/factcast/factus/FactusImpl.java (hunk 1 above)
Comment (uweschaefer) [reply factcast-factus/src/main/java/org/factcast/factus/FactusImpl.java:311]: will do

**Pull request merged**